src/mapper.rs

```diff
@@ -52,3 +52,13 @@ pub trait Mapper {
     /// ```
     fn unmap(&mut self, virt_start: usize, bytes: usize);
 }
+
+/// an identity mapper which maps a physical address into itself.
+#[derive(Clone, Copy, Debug)]
+pub struct Identity;
+impl Mapper for Identity {
+    unsafe fn map(&mut self, phys_base: usize, _bytes: usize) -> NonZeroUsize {
+        NonZeroUsize::new_unchecked(phys_base)
```

Comment:
Could you instead use `NonZeroUsize::new` to panic if the address 0 is passed?

Reply:
Used `.expect()` instead of asserting.